### PR TITLE
seq_cutoff: Allocate seq cutoff structures dynamically per core

### PR DESCRIPTION
**Pull request by robertbaldyga**
Signed-off-by: Robert Baldyga <robert.baldyga@intel.com>

### CODE REVIEW

File: src/ocf_seq_cutoff.c [hunk 1]
```diff
@@ -60,25 +60,37 @@ static struct ocf_rb_node *ocf_seq_cutoff_stream_list_find(
...
+		ocf_rb_tree_insert(&core->seq_cutoff->tree, &stream->node);
+		list_add_tail(&stream->list, &core->seq_cutoff->lru);
 	}
+
+	return 0;
+}
+
+void ocf_core_seq_cutoff_deinit(ocf_core_t core)
+{
+	env_vfree(core->seq_cutoff);
```
Comment (arutk) [inline src/ocf_seq_cutoff.c:93]: env_rwlock_destroy(&core->seq_cutoff->lock) is missing

Comment (robertbaldyga) [reply src/ocf_seq_cutoff.c:94]: Done.

File: src/ocf_seq_cutoff.c [hunk 2]
```diff
@@ -60,25 +60,38 @@ static struct ocf_rb_node *ocf_seq_cutoff_stream_list_find(
...
+		ocf_rb_tree_insert(&core->seq_cutoff->tree, &stream->node);
+		list_add_tail(&stream->list, &core->seq_cutoff->lru);
 	}
+
+	return 0;
+}
+
+void ocf_core_seq_cutoff_deinit(ocf_core_t core)
+{
+	env_rwlock_destroy(&core->seq_cutoff->lock);
```
Comment (arutk) [inline src/ocf_seq_cutoff.c:93]: I think checking whether seq_cutoff is NULL is required here (ocf_core_seq_cutoff_deinit might be called from _ocf_mngt_close_all_uninitialized_cores due to allocation failure)

Comment (robertbaldyga) [reply src/ocf_seq_cutoff.c:93]: Fixed.

**Review by arutk — APPROVED** [review]

**Review by mmichal10 — APPROVED** [review]

**Pull request merged**